stripe/src/test/java/com/stripe/android/paymentsheet/PaymentOptionsActivityTest.kt

```diff
@@ -122,6 +129,95 @@ class PaymentOptionsActivityTest {
         }
     }
 
+    @Test
+    fun `Verify Ready State`() {
+        val scenario = activityScenario()
+        scenario.launch(
+            createIntent(emptyList())
+        ).onActivity { activity ->
+            viewModel._viewState.value = ViewState.PaymentOptions.Ready
+
+            idleLooper()
+
+            val addBinding = PrimaryButtonBinding.bind(activity.viewBinding.addButton)
+
+            assertThat(addBinding.confirmedIcon.isVisible)
+                .isFalse()
+
+            assertThat(addBinding.label.text)
+                .isEqualTo("Add")
+
+            activity.finish()
+        }
+    }
+
+    @Test
+    fun `Verify Confirming state`() {
+        val scenario = activityScenario()
+        scenario.launch(
+            createIntent(emptyList())
+        ).onActivity { activity ->
+            viewModel._viewState.value = ViewState.PaymentOptions.Confirming
+
+            idleLooper()
+
+            val addBinding = PrimaryButtonBinding.bind(activity.viewBinding.addButton)
+
+            assertThat(addBinding.label.text)
+                .isEqualTo(activity.getString(R.string.stripe_paymentsheet_button_processing))
+        }
+    }
+
+    @Test
+    fun `Verify Successfully completed state closes the sheet`() {
+        val scenario = activityScenario()
+        scenario.launch(
+            createIntent(emptyList())
+        ).onActivity { activity ->
+            val paymentSelectionMock: PaymentSelection = mock()
+            viewModel._viewState.value = ViewState.PaymentOptions.Completed(
+                PaymentOptionResult.Succeeded(
+                    paymentSelectionMock
+                )
+            )
+            idleLooper()
+
+            // wait animate time...
+            runBlocking {
+                // wait for animation to complete
+                delay(1600)
```

Comment:
This could optionally reference `PrimaryButtonAnimator.HOLD_ANIMATION_ON_SLIDE_IN_COMPLETION`

Reply:
yeah, I will just make that internal